docs/book/core-concepts.md

```diff
@@ -152,6 +154,10 @@ zenml stack register STACK_NAME \
     -o ORCHESTRATOR_NAME
 ```
 
+**Container Registry**
+
+Container registries are where pre-built (Docker) containers are hosted, ready for use in various configurations. A ZenML workflow involving a container registry would see you spinning up a Kubernetes cluster and then deploying a pipeline to be run on Kubeflow Pipelines. As part of the deployment to the cluster, the ZenML base image would be downloaded and used as the basis for the deployed 'run'. When you are running a local Kubeflow stack, you will also have a local container registry which stores the container images you create that bundle up your pipeline code.
```

Comment:
Probably makes sense to make this a bit more general. A container registry is basically a store for docker images. The local variant of this is used to push a *new* docker image that is created based off a base image  or custom image of your choice. However in the future there can be a cloud variant which points to hosted registries for example the Google container registry or dockerhub

Reply:
The *newly* created images are pushed to the container registry specified in the stack, this however is not limited to a local registry (even now, it is entirely possible to specify a gcr.io/... registry URL as long as the local docker client is authorized to push to that registry).